release_dashboard/lib/widgets/common/checkbox_substep.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+typedef ClickCallback = void Function(String substepName);
+
+/// A checkbox that toggles between checked and unchecked upon clicking.
+///
+/// [clickCallback] is called every time the checkbox is clicked.
+class CheckboxAsSubstep extends StatelessWidget {
+  const CheckboxAsSubstep({
+    Key? key,
+    required this.isEachSubstepChecked,
+    required this.clickCallback,
+    required this.substepName,
+    this.subtitle,
+  }) : super(key: key);
+
+  final Map<String, bool> isEachSubstepChecked;
+  final ClickCallback clickCallback;
+  final String substepName;
+  final Widget? subtitle;
```

Comment:
subtitle could be any widget, to prepare for URL launcher, selectable text, and more.